posixlib/src/main/scala/scala/scalanative/posix/sys/wait.scala

```diff
@@ -0,0 +1,116 @@
+package scala.scalanative
+package posix
+package sys
+
+import scalanative.unsafe._
+
+import scalanative.posix.signal
+
+/** POSIX wait.h for Scala
+ *
+ *  The Open Group Base Specifications
+ *  [[https://pubs.opengroup.org/onlinepubs/9699919799 Issue 7, 2018]] edition.
+ *
+ *  A method with an XSI comment indicates it is defined in extended POSIX
+ *  X/Open System Interfaces, not base POSIX.
+ *
+ *  Note well: It is neither expect nor obvious from the declaration that the
+ *  wait() method of this class can conflict with Object.wait(Long). This makes
+ *  declaration and usage more difficult.
+ *
+ *  The simplest approach is to avoid "wait(Ptr[CInt])" and use the directly
+ *  equivalent idiom: // import scala.scalanative.posix.sys.wait.waitpid // or
+ *  sys.wait._ // Replace Ptr[CInt] with your variable. val status = waitpid(-1,
+ *  Ptr[CInt], 0)
+ *
+ *  If that approach is not available, one can try the following idiom: //
+ *  import scalanative.posix.sys.{wait => Wait} // import
+ *  scalanative.posix.sys.wait._ // for WIFEXITED etc. // Replace Ptr[CInt] with
+ *  your variable. val status = Wait.wait(Ptr[CInt])
+ */
+@extern
+object wait {
+  type id_t = types.id_t
+  type pid_t = types.pid_t
+
+  type sigval = signal.sigval
+  type siginfo_t = signal.siginfo_t
+
+  /* The type idtype_t shall be defined as an enumeration type whose possible
+   * values shall include at least the following: P_ALL P_PGID P_PID
+   */
+  type idtype_t = Int // POSIX enumeration in simple Scala common to 2.n & 3.n
+  @name("scalanative_posix_p_all")
```

Comment:
Eric,  thank you for the review.  Just in time, I was making a fix to not run on Windows.

The referenced section is entire unclear, at least to me if the "foo" part of "scalanative_foo_"
has to always be "c" or if it can be "posix" say for posix lib.   I can understand the c part for
clib but it seems out of place for posixlib.

I can change "_posix_" to "_c_" using the wonders of emacs.  A number of other ".c"
files use no "_foo_" at all, I think "_c_" is better than nothing.

Please advise.

Reply:
The example above for `P_ALL` would be `@name("scalanative_p_all")`. The `c_issock` you are seeing is because the macro is `C_ISSOCK` so maybe that example is a bit confusing thinking it is for a `C` library. I should have stated the rule in the docs. It should be `scalanative_` + `macro-name-in-lowercase` with no extra namespace.